packages/server/src/adapters/aws-lambda.ts

```diff
@@ -0,0 +1,80 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { URLSearchParams } from 'url';
+import { resolveHTTPResponse } from '..';
+import { HTTPHeaders, HTTPRequest } from '../http/internals/types';
+import { AnyRouter, inferRouterContext } from '../router';
+import type { AWSLambdaOptions, LambdaContext } from './lambda-utils';
+
+export type { CreateLambdaContextOptions } from './lambda-utils';
+
+function lambdaEventToHTTPRequest(event: APIGatewayProxyEvent): HTTPRequest {
+  const query = new URLSearchParams();
+  for (const [key, value] of Object.entries(
+    event.multiValueQueryStringParameters ?? {},
+  )) {
+    for (const v of value ?? []) {
+      query.append(key, v);
+    }
+  }
+  return {
+    method: event.httpMethod,
+    query,
+    headers: event.headers,
+    body: event.body,
+  };
+}
+
+function transformHeaders(
+  headers: HTTPHeaders,
+): APIGatewayProxyResult['headers'] {
+  const obj: APIGatewayProxyResult['headers'] = {};
+
+  for (const [key, value] of Object.entries(headers)) {
+    if (typeof value === 'undefined') {
+      continue;
+    }
+    obj[key] = Array.isArray(value) ? value.join(', ') : value;
+  }
+  return obj;
```

Comment:
Nit: not sure if mutating objects is the convention in this repo but usually `Object.fromEntries` is preferred these days:
```suggestion
  return Object.fromEntries(
    Object.entries(headers)
      .filter((_name, value]) => typeof value !== 'undefined') 
      .map(([name, value]) => Array.isArray(value) ? value.join(',') : value)
  );
```

(not sure if excluding `undefined` is strictly necessary, if not the `.filter` line can be deleted)

Reply:
None of the other adapters actually create an object for headers. The current way of mutating objects reads the closest to other adapters though.